### PR TITLE
done

### PR DESCRIPTION
**Pull request by Nebn1x**

### CODE REVIEW

File: src/main/java/springboot/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import springboot.dto.BookDto;
+import springboot.dto.CreateBookRequestDto;
+import springboot.service.BookService;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/springboot/controller/BookController.java:17]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/springboot/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class BookController {
+
```
Comment (Elena-Bruyako) [inline src/main/java/springboot/controller/BookController.java:19]: remove empty line

File: src/main/java/springboot/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.mapstruct.Mapper;
+import springboot.config.MapperConfig;
+import springboot.dto.BookDto;
+import springboot.dto.CreateBookRequestDto;
+import springboot.model.Book;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
```
Comment (Elena-Bruyako) [inline src/main/java/springboot/mapper/BookMapper.java:11]: remove empty line (if you add it - let's add everywhere, be consistent)

File: src/main/java/springboot/model/Book.java [hunk 4]
```diff
@@ -7,11 +7,9 @@
 import jakarta.persistence.Id;
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
-import lombok.Getter;
-import lombok.Setter;
+import lombok.Data;
 
-@Getter
-@Setter
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/springboot/model/Book.java:12]:
```suggestion
@Getter
@Setter
```

File: src/main/java/springboot/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -1,52 +1,59 @@
...
 
     @Override
-    public Book save(Book book) {
-        Session session = null;
-        Transaction transaction = null;
-        try {
-            session = sessionFactory.openSession();
-            transaction = session.beginTransaction();
-            session.persist(book);
+    public BookDto createBook(CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/springboot/repository/impl/BookRepositoryImpl.java:25]: revert changes

File: src/main/java/springboot/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -12,14 +16,26 @@
 public class BookServiceImpl implements BookService {
 
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto createBook(CreateBookRequestDto requestDto) {
+        return bookRepository.createBook(requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/springboot/service/impl/BookServiceImpl.java:23]: use mapper here to map dto to book

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/springboot/repository/impl/BookRepositoryImpl.java (hunk 5 above)
Comment (Nebn1x) [reply src/main/java/springboot/repository/impl/BookRepositoryImpl.java:25]: Return the standard save method, and in other classes, use createBook, inside which save from BookRepository will be called?

File: src/main/java/springboot/service/impl/BookServiceImpl.java (hunk 6 above)
Comment (Nebn1x) [reply src/main/java/springboot/service/impl/BookServiceImpl.java:23]: here should I use the save method that returns Book or createBook ?

File: src/main/java/springboot/model/Book.java (hunk 4 above)
Comment (Nebn1x) [reply src/main/java/springboot/model/Book.java:12]: why ?

Comment (Elena-Bruyako) [reply src/main/java/springboot/model/Book.java]: https://medium.com/@miguelangelperezdiaz444/the-hidden-dangers-of-lombok-annotations-in-your-java-code-what-you-need-to-know-8acdce2d6b89

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**